### PR TITLE
Trx payments

### PR DESCRIPTION
**Pull request by otselnik**

### CODE REVIEW

File: proxy/plugin/solana_rest_api.py [hunk 1]
```diff
@@ -37,6 +37,27 @@
...
 
+class PermanentAccounts:
+    def __init__(self, client, signer, proxy_id):
+        self.operator = signer.public_key()
+        self.operator_token = getTokenAddr(self.operator)
+
+        proxy_id_bytes = proxy_id.to_bytes((proxy_id.bit_length() + 7) // 8, 'big')
+        signer_public_key_bytes = bytes(signer.public_key())
+
+        storage_seed = shake_256(b"storage" + proxy_id_bytes + signer_public_key_bytes).hexdigest(16)
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api.py:48]: Why do you need to use `signer_public_key_bytes` when generating a storage_seed?
The `create_account_with_seed` function already uses them when generating the address.

File: proxy/plugin/solana_rest_api.py [hunk 2]
```diff
@@ -37,6 +37,27 @@
...
+        storage_seed = bytes(storage_seed, 'utf8')
+        self.storage = create_account_with_seed(client, funding=signer, base=signer, seed=storage_seed, storage_size=STORAGE_SIZE)
+
+        holder_seed = shake_256(b"holder" + proxy_id_bytes + signer_public_key_bytes).hexdigest(16)
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api.py:52]: Similar to `storage_seed` generation.

File: proxy/plugin/solana_rest_api_tools.py [hunk 3]
```diff
@@ -818,16 +962,38 @@ def getLamports(client, evm_loader, eth_acc, base_account):
     (account, nonce) = ether2program(bytes(eth_acc).hex(), evm_loader, base_account)
     return int(client.get_balance(account, commitment=Confirmed)['result']['value'])
 
-def getTokens(client, evm_loader, eth_acc, base_account):
+def getTokens(client, acc, evm_loader, eth_acc, base_account):
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: The `acc` name is very generic. It is worth coming up with a name that clearly reflects the purpose of the account.

File: proxy/run-proxy.sh [hunk 4]
```diff
@@ -69,5 +69,32 @@ fi
...
+# if [ -z "$COLLATERAL_POOL_BASE" ]; then
+#   echo "COLLATERAL_POOL_BASE is unset or set to the empty string. The pre-deployed token mint will be used"
+#   export COLLATERAL_POOL_BASE="$COLLATERAL_POOL_BASE_TEST_NET_ID"
+# else
+  if [ "$COLLATERAL_POOL_BASE" == "deploy" ]; then
+    echo "COLLATERAL_POOL_BASE is set to create. A new collateral pool accounts will be created. Creating accounts..."
+    #generate collateral pool accounts
+    solana -k /spl/bin/collateral-pool-keypair.json airdrop 1000
+    python3 /spl/bin/collateral_pool_generator.py /spl/bin/collateral-pool-keypair.json
+    export COLLATERAL_POOL_BASE="4sW3SZDJB7qXUyCYKA7pFL8eCTfm3REr8oSiKkww7MaT"
```
Comment (s-medvedev) [inline proxy/run-proxy.sh]: It's better to use the output from `solana-keygen pubkey -f /spl/bin/collateral-pool-keypair.json`.

File: proxy/plugin/solana_rest_api.py (hunk 1 above)
Comment (otselnik) [reply proxy/plugin/solana_rest_api.py:48]: https://github.com/neonlabsorg/proxy-model.py/pull/136

File: proxy/plugin/solana_rest_api.py (hunk 2 above)
Comment (otselnik) [reply proxy/plugin/solana_rest_api.py:52]: https://github.com/neonlabsorg/proxy-model.py/pull/136

File: proxy/plugin/solana_rest_api_tools.py (hunk 3 above)
Comment (otselnik) [reply proxy/plugin/solana_rest_api_tools.py]: Renamed to `signer`.

File: proxy/run-proxy.sh (hunk 4 above)
Comment (otselnik) [reply proxy/run-proxy.sh]: fixed

File: proxy/plugin/solana_rest_api.py (hunk 1 above)
Comment (s-medvedev) [reply proxy/plugin/solana_rest_api.py:48]: For this issue, it is enough to specify only `proxy_id_bytes`. But let's fix it as a separate task, not in this PR.

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**